RELEASE_PROCESS.md

```diff
@@ -0,0 +1,200 @@
+
+## Prerequisites for doing any release
+
+In order to do a release, you absolutely must have:
+
+* Commit access to the `ponyc` repo
+
+While not strictly required, you life will be made much easier if you:
+
+* Have a [bintray account](https://bintray.com/) and have been granted access `pony-language` organization by a "release admin".
+* Have read and write access to the ponylang [travis-ci](https://travis-ci.org) account
+* Have read and write access to the ponylang [appveyor](https://www.appveyor.com) account
+
+## Prerequisites for specific releases
+
+Before getting started, you will need a number for the version that you will be releasing as well as an agreed upon "golden commit" that will form the basis of the release.  Any commit is eligible to be a "golden commit" so long as:
+
+* It passed all CI checks
+* We were able to build "ponyc-master binaries" for it.
+
+### Validating "ponyc-master binaries"
+
+Currently, we validate binaries for RPM, Deb and Windows packaging. Follow the bintray links below to verify each of the package. You are looking under "Versions" for a successfully created package for the "golden commit". In the case of RPM and Deb, the versions look something like:
+
+`0.3.1-1885.8a8ee28`
+
+where the 
+
+`8a8ee28`
+
+is the sha for the commit we built from.
+
+For windows, version look something like:
+
+`ponyc-master-0.3.1-1526.8a8ee28`
+
+where the 
+
+`8a8ee28`
+
+is the sha for the commit we built from.
+
+* [RPM](https://bintray.com/pony-language/ponyc-rpm/ponyc-master)
+* [Debian](https://bintray.com/pony-language/ponyc-debian/ponyc-master)
+* [Windows](https://bintray.com/pony-language/ponyc-win/ponyc-master)
+
+### Validate external services are functional
+
+We rely on both Travis CI and Appveyor as part of our release process. Both need to be up and functional in order to do a release. Check the status of each before starting a release. If either is reporting issues, push the release back a day or until whenever they both are reporting no problems.
+
+* [Travis CI Status](https://www.traviscistatus.com)
+* [Appveyor Status](https://appveyor.statuspage.io)
+
+### A GitHub issue exists for the release
+
+All releases should have a corresponding issue in the [ponyc repo](https://github.com/ponylang/ponyc/issues). This issue should be used to communicate the progress of the release (updated as checklist steps are completed). You can also use the issue to notify other pony team members of problems.
+
+## Releasing
+
+For the duration of this document, we will pretend the "golden commit" version is `8a8ee28`, the version is `0.3.1`, and our release date is 2016-01-15. Any place you see those values, please substitute your own version.
+
+### Update the GitHub issue
+
+Leave a comment on the GitHub issue for this release to let everyone know that you are starting.
+
+### Update VERSION and CHANGELOG
+
+* git pull
+* git checkout -b release-0.3.1
```

Comment:
This is wrong. It needs to be updated to make sure we check out from out golden commit